Date handling cleanup

jamesagnew opened the pull request.

frankjtao reviewed (changes requested):
There is no issue on the SearchBuilder. But the some test cases are failed if  the Legacy flag is turned on for the the test `FhirResourceDaoR4FilterTest`

jamesagnew commented:
@frankjtao good call - I've fixed those issues and added `FhirResourceDaoR4FilterLegacySearchBuilderTest`, which is a copy of FhirResourceDaoR4FilterBuilderTest but with the legacy flag set to true
